types/underscore/index.d.ts

```diff
@@ -5030,7 +5030,7 @@ declare module _ {
         value(): V;
     }
 
-    interface _Chain<T, V = T> {
+    interface _Chain<T, V = T[]> {
```

Comment:
I did this because when I swapped out `_ChainIteratee` I ended up with a few errors in my company's solution around array proxy functions, which haven't had their return types updated to better reflect the `V = T` change that was made in #36230. It was also nice because it allowed simplification of several `_Chain` return types.

Doing this earlier (but after [this change](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/45304/files#diff-bc995fed3367d28165753a2581d1af54R4126)) would admittedly likely have been a quicker way to have fixed #36308 for a lot of functions since it better reflects the pre-#36230 `_Chain.value()` return type of `T[]`.

Reply:
It's also probably worth noting that I hadn't originally intended to simplify `_Chain` return types but linting was displeased when I didn't do so (and I'm admittedly not sad that it did).

Reply:
It does seem to remove a lot of cruft!